packages/components/nodes/memory/DynamoDb/DynamoDb.ts

```diff
@@ -102,13 +103,18 @@ const initializeDynamoDB = async (nodeData: INodeData, options: ICommonObject):
     const accessKeyId = getCredentialParam('accessKey', credentialData, nodeData)
     const secretAccessKey = getCredentialParam('secretAccessKey', credentialData, nodeData)
 
-    const config: DynamoDBClientConfig = {
-        region,
-        credentials: {
+    let credentials: DynamoDBClientConfig['credentials'] | undefined;
+    if (accessKeyId && secretAccessKey) {
+        credentials = {
             accessKeyId,
             secretAccessKey
-        }
+        };
     }
+    
+    const config: DynamoDBClientConfig = {
+        region,
+        credentials
```

Comment:
Curious if the credentials is undefined, what do we use? IAM role specified from your local env variables?

Reply:
Good question! The AWS SDK checks a series of different sources for credentials, in a specific order. You can find more information in the [AWS docs](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-credentials-node.html#credchain).

Passing the credentials through code is the way that takes precedence, but by making it optional, we allow users to choose any other preferred authentication method. In my case, that's using an IAM OIDC provider inside the K8S cluster. But there are many other methods.

I used the S3 file component code as a reference for my changes.

Reply:
Also, I'm going to open an issue to discuss how to add AWS authentication to the OpenSearch Vector Store component: [OpenSearch Docs](https://opensearch.org/docs/latest/clients/javascript/index/#authenticating-with-amazon-opensearch-service-aws-signature-version-4). Because I'm not sure if we should add it as an option to the existing component or create a new one called AWS OpenSearch.